frontends/mit-open/src/pages/OnboardingPage/OnboardingPage.tsx

```diff
@@ -112,45 +115,186 @@ const Label = styled.div({
   margin: "0 0 40px",
 })
 
+const profileSchema = yup.object().shape({
+  topic_interests: yup.array().of(yup.string()),
+  goals: yup.array().of(yup.string()),
+  certificate_desired: yup.string(),
+  current_education: yup.string(),
+  time_commitment: yup.string(),
+  learning_format: yup.string(),
+})
+
+const GRID_PROPS = {
+  justifyContent: "left",
+  columns: {
+    lg: 15,
+    md: 9,
+    xs: 3,
+  },
+  maxWidth: "lg",
+}
+const GRID_ITEM_PROPS = { xs: 3 }
+const GRID_STYLE_PROPS = {
+  gridProps: GRID_PROPS,
+  gridItemProps: GRID_ITEM_PROPS,
+}
+
 const OnboardingPage: React.FC = () => {
-  const [updates, setUpdates] = React.useState<PatchedProfileRequest>({})
   const { data: profile, isLoading: isLoadingProfile } = useProfileMeQuery()
+  const formId = useId()
+  const initialFormData = useMemo(() => {
+    return {
+      ...profile,
+      topic_interests:
+        profile?.topic_interests?.map((topic) => String(topic.id)) || [],
+    }
+  }, [profile])
   const { isLoading: isSaving, mutateAsync } = useProfileMeMutation()
   const [activeStep, setActiveStep] = React.useState<number>(0)
   const navigate = useNavigate()
-
-  const handleNext = () => {
-    // TODO: handle this error
-    mutateAsync(updates).then(() => {
-      setActiveStep((prevActiveStep) => prevActiveStep + 1)
-    })
-  }
-  const handleFinish = () => {
-    mutateAsync(updates).then(() => {
-      navigate(DASHBOARD)
-    })
-  }
+  const formik = useFormik({
+    enableReinitialize: true,
+    initialValues: initialFormData ?? profileSchema.getDefault(),
+    validationSchema: profileSchema,
+    onSubmit: async (values) => {
+      if (formik.dirty) {
+        await mutateAsync({
+          ...values,
+          topic_interests: values.topic_interests.map((id) => parseInt(id)),
+        })
+      }
+      if (activeStep < NUM_STEPS - 1) {
+        setActiveStep((prevActiveStep) => prevActiveStep + 1)
+      } else {
+        navigate(DASHBOARD)
+      }
+    },
+    validateOnChange: false,
+    validateOnBlur: false,
+  })
 
   const handleBack = () => {
     setActiveStep((prevActiveStep) => prevActiveStep - 1)
   }
 
-  const handleUpdate: ProfileFieldUpdateFunc = <
-    T extends keyof PatchedProfileRequest,
-  >(
-    name: T,
-    value: PatchedProfileRequest[T],
-  ) => {
-    if (!isEqual(updates[name], value)) {
-      setUpdates({
-        [name]: value,
-      })
-    }
-  }
-  if (typeof profile === "undefined") {
+  const { data: topics } = useLearningResourceTopics({ is_toplevel: true })
+  const topicChoices =
+    topics?.results?.map((topic) => ({
+      label: topic.name,
+      value: topic.id.toString(),
+    })) ?? []
+
+  if (!profile) {
     return null
   }
 
+  const pages = [
```

Comment:
`pages` could be a unction I guess, `getPage`.

Main changes:
- Single <form />` for onboarding.
- "Next" button is a submit button
- No more `React.useState<PatchedProfileRequest>({})`, since formik handles the updates
- Replaced all fields with plain `CheckboxChoiceBoxField`, `RadioChoiceBoxField`, `SimpleSelectField` instances
    - As long as all checkboxes share a common name, formik handles bundling the value into an array... nice.

This was a little bit quick and dirty. My thought is that onboarding and dashboard would share some of this:
- share constants like `GOALS_CHOICES` .... but not fields, since dashboard uses different fields for some of these
- maybe share a hook that handles initializing formik? Not that it is too complicated.